api/patientDataController.go

```diff
@@ -36,11 +36,27 @@ func (a *API) getDataV2(ctx context.Context, res *common.HttpResponseWriter) err
 	endDate := query.Get("endDate")
 	withPumpSettings := query.Get("withPumpSettings") == "true"
 	sessionToken := getSessionToken(res)
-	err := a.patientData.GetData(ctx, userID, res.TraceID, startDate, endDate, withPumpSettings, sessionToken, &buffer)
+	bgUnit := query.Get("bgUnit")
+	if bgUnit != usecase.MgdL && bgUnit != usecase.MmolL {
+		bgUnit = ""
+	}
+	getDataArgs := usecase.GetDataArgs{
+		Ctx:                        ctx,
```

Comment:
passing context in structs is not recommended 
https://go.dev/blog/context-and-structs#Storing%20Context%20in%20Structs%20Leads%20to%20Confusion